apps/explorer/src/pages/validators/Validators.tsx

```diff
@@ -17,16 +17,127 @@ import { Banner } from '~/ui/Banner';
 import { Card } from '~/ui/Card';
 import { Heading } from '~/ui/Heading';
 import { ImageIcon } from '~/ui/ImageIcon';
-import { ValidatorLink } from '~/ui/InternalLink';
+import { Link } from '~/ui/Link';
 import { PlaceholderTable } from '~/ui/PlaceholderTable';
 import { Stats } from '~/ui/Stats';
 import { TableCard } from '~/ui/TableCard';
 import { TableHeader } from '~/ui/TableHeader';
 import { Text } from '~/ui/Text';
 import { getName } from '~/utils/getName';
 
+const APY_DECIMALS = 4;
+
 const ValidatorMap = lazy(() => import('../../components/node-map'));
 
+function validatorsTableData(validators: ActiveValidator[], epoch: number) {
+    return {
+        data: validators.map((validator, index) => {
+            const validatorName = getName(
+                validator.fields.metadata.fields.name
+            );
+            return {
+                number: index + 1,
+                name: validatorName,
+                stake:
+                    +validator.fields.delegation_staking_pool.fields
+                        .sui_balance + +validator.fields.stake_amount,
+                apy: calculateAPY(validator, epoch),
+                commission: +validator.fields.commission_rate,
+                address: validator.fields.metadata.fields.sui_address,
+                lastEpoch:
+                    validator.fields.delegation_staking_pool.fields
+                        .rewards_pool,
+            };
+        }),
+        columns: [
+            {
+                header: '#',
+                accessorKey: 'number',
+                cell: (props: any) => (
```

Comment:
Don't love the `any`'s here. We can follow-up later with an improved Table component, just want to call it out